lib/model.js

```diff
@@ -654,6 +654,27 @@ var Model = define([QueryPlugin, Middleware], {
             }
         },
 
+        eager: function(associations) {
+            var model = new this(),
+                self = this,
+                includes = [],
+                associationsObj = {};
+
+            if (Array.isArray(associations)) {
+                includes = includes.concat(associations);
+            } else if(associations) {
+
+                includes.push(associations);
+            }
+
+            // change [] to {}
+            includes.forEach(association => {
+                associationsObj[association] = null;
+            });
+
+            model.dataset = model.dataset.eager(associationsObj, this);
```

Comment:
Why do we have to reassign dataset cant we just return `model.dataset.eager(associationsObj)`?